core-bundle/src/Routing/Content/PageRouteProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+use Contao\CoreBundle\Routing\RouteFactory;
+use Contao\PageModel;
+use Symfony\Component\Routing\Route;
+
+class PageRouteProvider implements ContentRouteProviderInterface
+{
+    /**
+     * @var RouteFactory
+     */
+    private $routeFactory;
+
+    public function __construct(RouteFactory $routeFactory)
+    {
+        $this->routeFactory = $routeFactory;
+    }
+
+    /**
+     * @param PageModel $page
```

Comment:
Why a phpDoc instead of a type hint in the method signature?

Reply:
Because the interface does not know the type and that doesn't allow for a typed argument.